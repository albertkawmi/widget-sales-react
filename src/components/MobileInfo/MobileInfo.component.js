import React from 'react'
import './MobileInfo.css'

// TODO save dismissed state to localStorage
class MobileInfo extends React.Component {
  state = { dismissed: false }

  render() {
    const { dismissed } = this.state
    const className = `mobile-info ${dismissed ? 'dismissed' : ''}`
    return (
      <div {...this.props} className={className}>
        <div className="mobile-info__text">
          This app is tricky to use on smaller screens. You can scroll (or swipe) the table around to see more data. Tapping a row will select a client and show their sales data beneath the table.
        </div>
        <button
          className="mobile-info__dismiss"
          onClick={() => this.setState({ dismissed: true })}
        >
          OK, got it.
        </button>
      </div>
    )
  }
}

export default MobileInfo
