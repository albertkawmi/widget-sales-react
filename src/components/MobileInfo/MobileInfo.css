.mobile-info {
  display: none;
}

.mobile-info__dismiss {
  margin-top: 1rem;
  width: 100%;
}

@media (max-width: 600px) {
  .mobile-info {
    display: block;
    background: lemonchiffon;
    padding: 0.5rem;
    border: solid 1px #e8be1e;
  }
  .mobile-info.dismissed {
    display: none;
  }
}
