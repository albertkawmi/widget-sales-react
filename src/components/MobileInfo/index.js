import MobileInfo from './MobileInfo.component'
export default MobileInfo