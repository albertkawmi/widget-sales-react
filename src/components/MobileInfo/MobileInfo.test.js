/* eslint-env jest */
import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import MobileInfo from './MobileInfo.component'

Enzyme.configure({ adapter: new Adapter() });

describe('<MobileInfo />', () => {
  const mobileInfo = Enzyme.shallow(
    <MobileInfo />
  )

  it('renders a help message for mobile users', () => {
    expect(mobileInfo).toMatchSnapshot()
  })

  it('hides when dismissed', () => {
    expect(
      mobileInfo.find('.mobile-info.dismissed').exists()
    ).toBe(false)

    mobileInfo.find('.mobile-info__dismiss')
      .simulate('click')

    expect(
      mobileInfo.find('.mobile-info.dismissed').exists()
    ).toBe(true)
  })
})
