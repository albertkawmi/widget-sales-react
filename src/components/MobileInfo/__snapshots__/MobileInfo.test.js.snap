// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MobileInfo /> renders a help message for mobile users 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MobileInfo />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="mobile-info__text"
>
          This app is tricky to use on smaller screens. You can scroll (or swipe) the table around to see more data. Tapping a row will select a client and show their sales data beneath the table.
</div>,
        <button
          className="mobile-info__dismiss"
          onClick={[Function]}
>
          OK, got it.
</button>,
      ],
      "className": "mobile-info ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This app is tricky to use on smaller screens. You can scroll (or swipe) the table around to see more data. Tapping a row will select a client and show their sales data beneath the table.",
          "className": "mobile-info__text",
        },
        "ref": null,
        "rendered": "This app is tricky to use on smaller screens. You can scroll (or swipe) the table around to see more data. Tapping a row will select a client and show their sales data beneath the table.",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "OK, got it.",
          "className": "mobile-info__dismiss",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "OK, got it.",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="mobile-info__text"
>
            This app is tricky to use on smaller screens. You can scroll (or swipe) the table around to see more data. Tapping a row will select a client and show their sales data beneath the table.
</div>,
          <button
            className="mobile-info__dismiss"
            onClick={[Function]}
>
            OK, got it.
</button>,
        ],
        "className": "mobile-info ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This app is tricky to use on smaller screens. You can scroll (or swipe) the table around to see more data. Tapping a row will select a client and show their sales data beneath the table.",
            "className": "mobile-info__text",
          },
          "ref": null,
          "rendered": "This app is tricky to use on smaller screens. You can scroll (or swipe) the table around to see more data. Tapping a row will select a client and show their sales data beneath the table.",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "OK, got it.",
            "className": "mobile-info__dismiss",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "OK, got it.",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
